### PR TITLE
Big file (>2GB) support in vol_libs

### PR DESCRIPTION
**Pull request by capnramses**
**Reason for PR**

* Studio captures >2GB were not running in the Unity Plugin project. https://github.com/Volograms/volograms_unity_plugin
* Analysis showed that vol_libs was rejecting files of this size with an error message, and that underlying file size and seek functions used were not 64-bit offset aware on all platforms.

**Work Done in this PR**

* Functions `fseek()`, `ftell()` were replaced by fseeko(),ftello() on POSIX-compliant platforms, and Windows now uses `_fseeki64()`, `_ftelli64()`, which have equivalent functionality.
* A `#define` was introduced to differentiate platforms, which now in code refer to `vol_geom_fseeko` etc.
* Instead of using `fseek`/`ftell` for getting file size after opening the file, the functions `stat()/_stat64()` (Windows) are used, which does not require opening or seeking through large files.
* These functions are from `sys/stat.h` on both systems.
* Set `-D_FILE_OFFSET_BITS=64` in Makefile for POSIX builds, which enforces 64-bit `off_t` even on 32-bit systems (e.g. Nexus 6 or something). 
* Updated headers and definitions for Linux builds.
* Please note I also changed the default `vol_geom_size_t` type from `uint64_t` to `int64_t` to more closely match the above functions' offset types.
    * This may affect the Unity Plugin's library wrapper.
    * Because I have moved to a signed type, several places require a (type cast) to silence warnings of addition of bytes between the `vol_geom_size_t` and `size_t`.

**Testing Summary**

* Tested large captures with older C# player, to ensure they were playable and the files can work. (They worked).
* Tested on Windows with MinGW/GCC, and MSVC (2019) builds.
* Tested on Windows 10 with ld, sd, and hd captures from the "Ulysses" project. Before, and after this PR, using example programs in this library.
   * Smallest ld capture worked before the change, the others all worked after the changes in this PR.
   * Initial load times for very large files can be a few seconds.
   * Run-time memory footprint for ~8GB HD files is <60MB, and performance on desktop is great, using "streaming" mode. CPU usage on my desktop is very low. This is really good from a low-perf device point of view.
   * Run-time memory footprint for HD with 2k textures is ~90MB main memory. Also very low CPU footprint, but fairly high GPU usage. Hardware video decoding is not used, and very little GPU memory is retained.
image
image
* Tested HD version on 64-bit Ubuntu (also used by GitHub Actions for running Tests).

**Risk Introduced by PR**

* *Regression risk of these changes is high*.
   * This library is a dependency of multiple projects and shipped products, so we'll need to be careful to test this very thoroughly, as we roll it out into other projects.
   * Testing on multiple platforms/compilers is essential.
   * Doing roll-out to other projects in stages (not all at once) is a good idea.
   * Update the Volu app last, to minimise risk to shipped product.
   * Any other very large files would be really handy to test with.
   
**ToDo (Before Merging)**

* Test on macOS ( **need a hand here!** )
   
**ToDo (After Merging)**

* Tag and Release 0.10.0
* Update Unity Plugin with latest vol_libs. (iOS and Android testing).
* Update Unreal Plugin with latest vol_libs.
* Update Volu app with latest vol_libs. (iOS and Android testing).
* Update vol_cl_tools with latest vol_libs.
* Lots of testing!

### CODE REVIEW

File: tests/geom_timed_test.c [hunk 1]
```diff
@@ -17,10 +17,11 @@
 #include <time.h>
 #ifdef _WIN32
 #include <windows.h> /* for time */
```
Comment (capnramses) [inline tests/geom_timed_test.c:19]: This was just a change in case anyone wants to compile this test program on Windows (not used by us in GitHub Actions).

File: tools/obj2vol/main.c [hunk 2]
```diff
@@ -54,7 +54,7 @@ static bool _load_obj( const char* filename ) {
   if ( !v_buff || !vn_buff || !vt_buff || !f_buff ) { return false; }
```
Comment (capnramses) [inline tools/obj2vol/main.c:55]: I forgot to make this multi-platform. It's a tiny tool to (badly) convert small .objs (like a cube) into volograms for testing.
The easy option here might be just to revert this change.

**Comment by capnramses** [comment]
I made an update - the 3 unique crashes found by the fuzzer were all due to switching from unsigned to signed size type (from uint64_t to int64_t) for offsets. With bad files when this produced negative integers for sizes etc it would crash. I fixed this by checking all fields memcopied from the file were >= 0.
I have re-fuzzed and found no crashes.

**Review by patrick-volograms — APPROVED** [review]
Tested on Mac the following: 
* Generated the tests using `make test_programs` and ran `./run_tests.sh`, all tests succeeded
* Generated the examples using `make example_programs` and run `vol_geom_av_opengl` example with 2 different large volograms (>2 GB) with 2k video, worked as expected

**Review by valeria-volu — APPROVED** [review]

**Comment by valeria-volu** [comment]
Code reviewed. Also tested on Mac - working well

**Comment by capnramses** [comment]
Thanks @valeria-volu and @patrick-volograms . I will update the ReadMe with feedback from questions and let you review before merging.

**Comment by capnramses** [comment]
alright i think the readme updates will be easier to fix up after merge so im going to go ahead and merge first then ask for a review
